demisto_sdk/commands/pre_commit/hooks/hook.py

```diff
@@ -24,6 +24,7 @@ def __init__(
         self.mode = mode
         self.all_files = all_files
         self.input_mode = input_mode
+        self._set_properties(hook={}, to_delete=to_delete)
```

Comment:
to_delete cant be used properly in the init method, since it would delete all the properties before the hook has a change to process them. We should get rid of the deleting from this function, and offer a utility function to delete fields after the logic is done

Please update the docker class to fit with this standard